activerecord/lib/active_record/attributes.rb

```diff
@@ -245,8 +246,17 @@ def define_attribute(
 
       def load_schema! # :nodoc:
         super
-        attributes_to_define_after_schema_loads.each do |name, (type, options)|
-          define_attribute(name, _lookup_cast_type(name, type, options), **options.slice(:default))
+        attributes_to_define_after_schema_loads.each do |name, (type, options, transform)|
+          if type.is_a?(Symbol)
+            adapter_name = ActiveRecord::Type.adapter_name_from(self)
+            type = ActiveRecord::Type.lookup(type, **options.except(:default), adapter: adapter_name)
+          elsif type.nil?
+            type = type_for_attribute(name)
+          end
+
+          type = transform[type] if transform
+
+          define_attribute(name, type, **options.slice(:default))
```

Comment:
```suggestion
          define_attribute(name, type, default: options[:default])
```

Since we're only passing one value, would it be clearer as just this?

Reply:
That would behave differently when there is no default.  When there is no default value, `**options.slice(:default)` is equivalent to `**{}`, but `default: options[:default]` is equivalent to `default: nil`, which would override [`NO_DEFAULT_PROVIDED`](https://github.com/rails/rails/blob/db7b7e8f3d30bb5f12988b49bb3d80c4441b64f1/activerecord/lib/active_record/attributes.rb#L239).